core/runtime/common/storage_code_provider.cpp

```diff
@@ -26,7 +26,8 @@ namespace kagome::runtime {
     BOOST_ASSERT(runtime_upgrade_tracker_ != nullptr);
     last_state_root_ = storage_->getRootHash();
     auto batch = storage_->getEphemeralBatch();
-    setStateCodeFromBatch(batch);
+    BOOST_ASSERT_MSG(batch.has_value(), "Error getting a batch of the storage");
```

Comment:
Possible, it should be logged as critical, because this situation is reason of crash in next line. Log will be useful for analize.